### PR TITLE
feat(core): add e2e to ci workflows

### PR DESCRIPTION
**Pull request by meeroslav**
## Current Behavior
Generated workflow has just `lint, test and build` defined

## Expected Behavior
If `@nx/cypress` or `@nx/playwright` are detected the workflow will also add `e2e`

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/workspace/src/generators/ci-workflow/ci-workflow.ts [hunk 1]
```diff
@@ -30,11 +29,19 @@ export async function ciWorkflowGenerator(host: Tree, schema: Schema) {
   if (!nxCloudUsed) {
     throw new Error('This workspace is not connected to Nx Cloud.');
   }
+  const packageJson = readJson(host, 'package.json');
+  const allDependencies = {
+    ...packageJson.dependencies,
+    ...packageJson.devDependencies,
+  };
+  const hasE2E =
+    allDependencies['@nrwl/cypress'] || allDependencies['@nrwl/playwright'];
```
Comment (AgentEnder) [inline packages/workspace/src/generators/ci-workflow/ci-workflow.ts:38]:
```suggestion
    allDependencies['@nx/cypress'] || allDependencies['@nx/playwright'];
```

**Review by AgentEnder — CHANGES_REQUESTED** [review]
We should be looking at @nx/* packages, not @nrwl/*

File: packages/workspace/src/generators/ci-workflow/ci-workflow.ts (hunk 1 above)
Comment (meeroslav) [reply packages/workspace/src/generators/ci-workflow/ci-workflow.ts:38]: Good catch. That was copilot being smart and I missed it.

**Review by AgentEnder — APPROVED** [review]

File: packages/workspace/src/generators/ci-workflow/files/gitlab/.gitlab-ci.yml__tmpl__ [hunk 2]
```diff
@@ -21,4 +21,4 @@ variables:
     - NX_BASE=${CI_MERGE_REQUEST_DIFF_BASE_SHA:-$CI_COMMIT_BEFORE_SHA}
 
     - <%= packageManagerPrefix %> nx-cloud record -- nx format:check --base=$NX_BASE --head=$NX_HEAD
-    - <%= packageManagerPrefix %> nx affected --base=$NX_BASE --head=$NX_HEAD -t=lint,test,build
+    - <%= packageManagerPrefix %> nx affected --base=$NX_BASE --head=$NX_HEAD -t=lint test build<% if(hasE2E){ %> e2e<% } %>
```
Comment (FrozenPandaz) [inline packages/workspace/src/generators/ci-workflow/files/gitlab/.gitlab-ci.yml__tmpl__:24]: e2e-ci

**Review by FrozenPandaz — APPROVED** [review]

File: packages/workspace/src/generators/ci-workflow/files/gitlab/.gitlab-ci.yml__tmpl__ (hunk 2 above)
Comment (meeroslav) [reply packages/workspace/src/generators/ci-workflow/files/gitlab/.gitlab-ci.yml__tmpl__]: Done

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**